u2f-ref-code/java/src/com/google/u2f/codec/RawMessageCodec.java

```diff
@@ -192,6 +194,80 @@ public static AuthenticateResponse decodeAuthenticateResponse(byte[] data) throw
     }
   }
 
+  public static TransferAccessResponse decodeTransferAccessResponse(byte[] data) 
+      throws U2FException {
+    final int SEQUENCE_NUMBER_SIZE = 1;
+    final int PUBLIC_KEY_SIZE = 65;
+    final int APP_ID_SIZE = 32;
+    
+	try {
+	  DataInputStream inputStream = new DataInputStream(new ByteArrayInputStream(data));
+	  byte controlByte = inputStream.readByte();
+	  
+	  if (!inputStream.markSupported()) {
```

Comment:
I'm not actually sure. I just wanted to make sure that if this is the reason the program failed that it was clear. Is this a bad reason to have the check?